src/utils/azureUtils.ts

```diff
@@ -18,6 +18,13 @@ import { treeUtils } from './treeUtils';
 export function createGroupConfigFromResource(resource: AppResource, subscriptionId: string | undefined): GroupingConfig {
     const id = nonNullProp(resource, 'id');
     const unknown = localize('unknown', 'Unknown');
+    let iconPath: TreeItemIconPath;
```

Comment:
Logic feels a little backwards to me here.  What about doing something like:

```
let iconPath = getIconPath(resource.azExtResourceType);
if (resource.azExtResourceType === AzExtResourceType.ContainerAppsEnvironment) {
    // comment as to why this is needed
        iconPath = getIconPath(AzExtResourceType.ContainerApps);
    }

Reply:
I think it's actually swapped, I removed this one and it didn't work, but I removed the iconPath change in the other section and it worked